lib/interfaces/bdd.js

```diff
@@ -36,7 +59,7 @@ const matchStep = (step) => {
     const expression = new CucumberExpression(stepName, parameterTypeRegistry);
     const res = expression.match(step);
     if (res) {
-      const fn = steps[stepName];
+      const fn = (featureName && Object.keys(matchingSteps).includes(stepName)) ? _findClosestPath(featureName, matchingSteps[stepName]) : steps[stepName];
```

Comment:
again, I can't understand this. Please move this logic to a separate function if this is repeated 2 times.